swatch-contracts/src/test/java/com/redhat/swatch/contract/resource/ContractsTestingResourceTest.java

```diff
@@ -115,4 +127,46 @@ void forceSyncForOrgShouldReturnSuccess() {
     resource.forceSyncSubscriptionsForOrg(ORG_ID);
     verify(subscriptionSyncService).forceSyncSubscriptionsForOrg(ORG_ID, false);
   }
+
+  @Test
+  void testSyncSkuErrorResult() {
+    doThrow(new NotFoundException()).when(offeringSyncService).syncOffering(any(String.class));
+    assertThrows(NotFoundException.class, () -> resource.syncOffering("TEST_SKU"));
+
+    doThrow(new ForbiddenException()).when(offeringSyncService).syncOffering(any(String.class));
+    assertThrows(ForbiddenException.class, () -> resource.syncOffering("TEST_SKU"));
+
+    doThrow(new BadRequestException()).when(offeringSyncService).syncOffering(any(String.class));
+    assertThrows(BadRequestException.class, () -> resource.syncOffering("TEST_SKU"));
+
+    when(offeringSyncService.syncOffering(any(String.class)))
+        .thenReturn(SyncResult.SKIPPED_NOT_FOUND);
+    assertThrows(NotFoundException.class, () -> resource.syncOffering("TEST_SKU"));
+
+    when(offeringSyncService.syncOffering(any(String.class)))
+        .thenReturn(SyncResult.SKIPPED_DENYLISTED);
+    assertThrows(ForbiddenException.class, () -> resource.syncOffering("TEST_SKU"));
+
+    ExternalServiceException ese = mock(ExternalServiceException.class);
```

Comment:
The ExternalServiceException is never thrown by the OfferingSyncService, only by the SubscriptionSyncService. 
As far I see, the intention here was to mock the ApiException which is thrown by the ProductApi client. This client should directly return an instance of ApiException.

Reply:
When I ran it with wiremock, this is exactly what showed up when the SKU was not found. If that is not a real live scenario, then what is? 
ExternalServiceException is a RuntimeException. I caught it and handled it for the reason above.

Reply:
Maybe we're not speaking about the same use case. I tried to sync an offering without using stubs, so an exception is expected because the ProductApi is down and I see the ProcessingException instead of the ExternalServiceException:
![Captura de pantalla de 2024-08-07 15-04-07](https://github.com/user-attachments/assets/c03d1273-7045-4a32-8673-bd1d6399fed4)